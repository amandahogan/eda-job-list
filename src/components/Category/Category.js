import React, { PropTypes } from 'react'
import { List } from 'semantic-ui-react'

import Tag from '../Tag'

class Category extends React.Component {
  render () {
    const { category, tags } = this.props
    return (
      <div>
        <span><strong>{category.name}: </strong></span>
        <List horizontal>
          {tags.map((tag, index) => {
            return (
              <List.Item key={index}>
                <List.Content>
                  <Tag category={category} value={tag.value} />
                </List.Content>
              </List.Item>
            )
          })}
        </List>
      </div>
    )
  }
}

Category.propTypes = {
  category: PropTypes.shape({
    colour: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired
  }).isRequired,
  tags: PropTypes.arrayOf(PropTypes.object).isRequired
}

export default Category
