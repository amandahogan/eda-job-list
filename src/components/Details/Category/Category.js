import React, { PropTypes } from 'react'
import { List } from 'semantic-ui-react'

import Tag from '../../Tag'
import AddTag from './AddTag'

class Category extends React.Component {
  render () {
    const { category, tags } = this.props
    return (
      <div>
        <span><strong>{category.name}: </strong></span>
        <List horizontal>
          {tags.map((tag, index) => {
            return (
              <List.Item key={index}>
                <List.Content>
                  <Tag
                    addTagValue={this.props.addTag}
                    category={category}
                    value={tag.value}
                  />
                </List.Content>
              </List.Item>
            )
          })}
        </List>
        <AddTag addTagValue={this.props.addTagValue} />
      </div>
    )
  }
}

Category.propTypes = {
  addTagValue: PropTypes.func.isRequired,
  category: PropTypes.shape({
    colour: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired
  }).isRequired,
  tags: PropTypes.arrayOf(PropTypes.object).isRequired
}

export default Category
