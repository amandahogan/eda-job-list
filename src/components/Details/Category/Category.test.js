import { shallow } from 'enzyme'
import React from 'react'

import Category from './Category'

describe('<Category />', () => {
  it('matches snapshot', () => {
    const category = {
      colour: 'green',
      name: 'Test'
    }
    const tags = []
    expect(
      shallow(
        <Category
          addTagValue={() => {}}
          category={category}
          tags={tags}
        />
      ).getNode()
    ).toMatchSnapshot()
  })
})
