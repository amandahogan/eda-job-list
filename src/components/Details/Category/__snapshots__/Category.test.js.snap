exports[`<Category /> matches snapshot 1`] = `
<div>
  <span>
    <strong>
      Test
      : 
    </strong>
  </span>
  <List
    horizontal={true}>
    
  </List>
</div>
`;
